server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -866,6 +887,15 @@ public boolean moveShards() {
                 // can use the cached decision.
                 final var moveDecision = shardMoved ? decideMove(index, shardRouting) : storedShardMovement.moveDecision();
                 if (moveDecision.isDecisionTaken() && moveDecision.cannotRemainAndCanMove()) {
+                    if (notPreferredLogger.isDebugEnabled()) {
+                        logMoveNotPreferred.maybeExecute(
```

Comment:
Can we do away with the throttled logging? Rather, just log everything.

Conceptually, we should only be picking the best shard to move away when a node is hot-spotting. We fix the hot-spot with one move, and then we don't have a hot-spot and this code doesn't run. So logging once per node per 30 seconds. I don't think it needs throttling?

Reply:
OK, removed in 3ba31a0a7457d1821642b23e5b25dff7aa2ebb46 🤞